### DISCUSSION TITLE
Issue: setRequestLocale behavior with Next.js 15 - Potential inconsistency?

### DISCUSSION THREAD

**Discussion by adilsonx** [amannn/next-intl#1563] [Q&A]
Hello,

I've noticed a potential issue with `setRequestLocale` in Next.js 15 that differs from the previous behavior with `unstable_setRequestLocale` in Next.js 14.

**Context:**
- Using Next.js 15
- Trying to achieve static rendering for some pages

**Current Behavior:**
Even though my pages show as `●  (SSG) prerendered as static HTML (uses generateStaticParams)` during build time, I still see the following behavior:
1. Console logs are still appearing during navigation (e.g., "FETCHING CONTENT FROM GITHUB")
2. To achieve proper static rendering, I need to add both `generateStaticParams` and `setRequestLocale` in:
   - The root layout
   - Any intermediate layouts
   - The page itself

**Previous Behavior (Next.js 14 with `unstable_setRequestLocale`):**
I only needed to add `generateStaticParams` and `unstable_setRequestLocale` to the pages I wanted to render statically, and it worked as expected.

**Questions:**
Is this the expected behavior with Next.js 15?

Let me know if you need any additional information or reproduction steps.

**Comment by amannn** [comment]
Hmm—no, I haven't encountered this so far. But I also haven't worked with Next.js 15 much due to [a bug](https://github.com/amannn/next-intl/issues/1442) I'm waiting for them to fix.

On a separate note, [the Next.js team is working on `rootParams`](https://github.com/amannn/next-intl/issues/663#issuecomment-2478719013)—finally the API to make `setRequestLocale` obsolete in the future. It's still early, but I could imagine that this integrates better with Next.js 15.
